src/code/z_map_data.c

```diff
@@ -0,0 +1,389 @@
+        { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 0, 0 },
+        { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 0, 0 },
+        { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 0, 0 },
+        { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 0, 0 },
+        { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 0, 0 },
+    },
+};
+
+static s16 roomCompassOffsetX[10][44] = {
+    { 1090, 1390, 1560, 1220, 1200, 1390, 1770, 1610, 2000, 1290, 1420, 1110, 1040, 470, 790,
+      1570, 720,  1000, 1580, 70,   0,    0,    0,    0,    0,    0,    0,    0,    0,   0,
+      0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0 },
+    { 940, 320, 1500, 240, 580, 1510, 720, 1030, 800, 660, 180, 520, 310, 550, 790, 1650, 1000, 1570, 80, 70, 0, 0,
+      0,   0,   0,    0,   0,   0,    0,   0,    0,   0,   0,   0,   0,   0,   0,   0,    0,    0,    0,  0,  0, 0 },
+    { 1130, 1070, 1090, 1160, 1500, 690, 1540, 920, 1160, 700, 1650, 950, 1380, 1460, 830, 1170, 1620, 0, 0, 0, 0, 0,
+      0,    0,    0,    0,    0,    0,   0,    0,   0,    0,   0,    0,   0,    0,    0,   0,    0,    0, 0, 0, 0, 0 },
+    { 1130, 1170, 965,  890,  1170, 1460, 1170, 800,  1320, 880, 1130, 1590, 1390, 830, 610,
+      580,  710,  980,  1640, 1510, 590,  1610, 1130, 1130, 820, 1320, 1620, 0,    0,   0,
+      0,    1300, 1270, 0,    0,    0,    0,    0,    0,    0,   0,    0,    0,    0 },
+    { 1160, 620,  1330, 1280, 440,  600,  810, 830,  720, 1170, 1490, 1640, 1870, 1800, 1610,
+      1130, 860,  1310, 1140, 850,  760,  380, 800,  800, 1930, 1410, 640,  845,  810,  810,
+      850,  1390, 1540, 1650, 1880, 1530, 420, 1950, 0,   0,    0,    0,    0,    0 },
+    { 1120, 1290, 1120, 1380, 930,  1520, 1980, 2010, 1590, 1510, 1500, 1300, 1240, 1800, 1290,
+      1450, 1560, 880,  820,  820,  1060, 1670, 1120, 1130, 1130, 1290, 1290, 1280, 1390, 940,
+      1520, 1520, 1980, 1620, 1510, 1490, 1240, 1290, 1450, 880,  880,  1060, 1670, 1520 },
+    { 800, 1500, 1370, 1730, 1590, 1020, 1060, 1470, 1600, 1830, 1630, 2000, 650,  660,  1020,
+      880, 940,  720,  570,  620,  570,  550,  970,  920,  1040, 1150, 1200, 1550, 1520, 1020,
+      820, 1010, 0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0 },
+    { 1320, 1320, 1090, 1510, 1480, 940, 920, 910, 800, 820,  1150, 1000, 1800, 1660, 1090,
+      1630, 710,  1670, 830,  770,  800, 850, 830, 820, 1800, 1090, 850,  0,    0,    0,
+      0,    0,    0,    0,    0,    0,   0,   0,   0,   0,    0,    0,    0,    0 },
+    { 1080, 1420, 1620, 1040, 940, 1190, 1310, 1090, 1380, 1080, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+      0,    0,    0,    0,    0,   0,    0,    0,    0,    0,    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 },
+    {
+        1070, 1180, 1270, 990, 1280, 1450, 1680, 1530, 760, 860, 1500, 800, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0,    0,    0,    0,   0,    0,    0,    0,    0,   0,   0,    0,   0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
```

Comment:
The last sub array has a trailing comma here but the other ones don't. I'm not sure which is better but I think it should be consistent.